### PR TITLE
Add basic readme and pre-commit

### PR DESCRIPTION
**Pull request by Mayank808**
## Notion ticket link

[Set up linting and formatting pre-commit hooks](https://www.notion.so/uwblueprintexecs/Set-up-linting-and-formatting-pre-commit-hooks-11410f3fb1dc80d6bd9ae7469cc0f283?pvs=4)

## Implementation description
* Setup basic pre-commit hooks commented out ones that are currently breaking/causing issues will look into them later/if needed
* Wrote quick startup guide in readme for backend

## Steps to test
1. run ./setup.sh and ensure pre-commit is installed properly (script can be removed and just add the pip install/brew install to read me)
2. pre-commit run or git commit -m/git push on changes
3. should lint and refactor changes if needed. (will fail if it requires any change)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: update_secret_files.py [hunk 1]
```diff
@@ -4,53 +4,67 @@
 # Open secret.config file
 configFileNotFound = False
 try:
-	configFile = open('secret.config')
```
Comment (mslwang) [inline update_secret_files.py:7]: Note: this looks like formatting output.

File: backend/README.md [hunk 2]
```diff
@@ -1 +1,74 @@
...
+
+# Add Flags if needed
+-d # Runs container in background
+```
+
+You can adjust the ports as needed. For example, 8080:8080 maps the container’s port 8080 to your local machine's port 8080. Once running, you should be able to access the backend locally via: -->
+
+The backend runs at http://localhost:8080 and the frontend runs at http://localhost:3000.
+
+## Environment Variables
```
Comment (mslwang) [inline backend/README.md:74]: Should add instructions to set up pre-commit hooks.

**Review by mslwang — APPROVED** [review]
LGTM functionality wise
[Precommit hook rejecting](https://app.warp.dev/block/EeisrHbcV5aayAyGqtddKX)
[Precommit hook accepting](https://app.warp.dev/block/zRhjQjDuJOvAHnucBMVGBP)

Would be great to just add some extra touches to the README, then we can merge.

**Review by EdmondLi1 — APPROVED** [review]
LGTM!

**Pull request merged**